tests/integration/test.replication.js

```diff
@@ -2057,163 +2057,135 @@ adapters.forEach(function (adapters) {
       for (var i = 0; i < num; i++) {
         docs.push({
           _id: 'doc_' + i,
-          foo: 'bar_' + i
+          foo: 'bar_' + i,
+          // needed to cause the code to fetch using get
+          _attachments: {
+            text: {
+              content_type: 'text\/plain',
+              data: "VGhpcyBpcyBhIGJhc2U2NCBlbmNvZGVkIHRleHQ="
+            }
+          }
         });
       }
+
       // Initialize remote with test documents
-      remote.bulkDocs({ docs: docs }, {}, function () {
-        var get = remote.get;
-        function first_replicate() {
-          // Mock remote.get to fail writing doc_3 (fourth doc)
-          remote.get = function () {
-            // Simulate failure to get the document with id 'doc_4'
-            // This should block the replication at seq 4
-            if (arguments[0] === 'doc_4') {
-              arguments[2].apply(null, [{}]);
-            } else {
-              get.apply(this, arguments);
-            }
-          };
-          // Replicate and confirm failure, docs_written and target docs
-          db.replicate.from(remote, function (err, result) {
-            should.exist(err);
-            should.exist(result);
-            result.docs_written.should.equal(4);
-            function check_docs(id) {
-              if (!id) {
-                second_replicate();
-                return;
-              }
-              db.get(id, function (err, exists) {
-                if (exists) {
-                  should.not.exist(err);
-                } else {
-                  should.exist(err);
+      // can't use bulkDocs because order is important
+      var results = [];
+      function save_doc(i) {
+        if (i >= docs.length) {
+          first_replicate();
+          return;
+        }
+        remote.put(docs[i])
+          .then(function(result) {
+            results.push(result);
+            save_doc(i + 1);
+          })
+          .catch(function(err) {
+            console.error('Error saving doc', err);
+          });
+      }
+      save_doc(0);
+
+      var bulkGet = remote.bulkGet;
+      function first_replicate() {
+        remote.bulkGet = function () {
```

Comment:
an issue here is that the order of results in _changes is not guaranteed. The test will fail against CouchDB 2.X and Cloudant because the result from bulkGet doesn't match the documents that were requested.